datafusion/functions-nested/src/planner.rs

```diff
@@ -148,6 +148,13 @@ impl ExprPlanner for FieldAccessPlanner {
             // expr[idx] ==> array_element(expr, idx)
             GetFieldAccess::ListIndex { key: index } => {
                 match expr {
+                    // Special case for accessing map value with non-string values
+                    Expr::ScalarFunction(scalar_func) if is_map(&scalar_func) => {
+                        Ok(PlannerResult::Planned(get_field_from_expr(
+                            Expr::ScalarFunction(scalar_func),
+                            *index,
```

Comment:
I think index is always literal, therefore we don't really need `get_field_from_expr`

Reply:
This was helpful. Thanks